### PR TITLE
Create initial version on VueJS

### PR DESCRIPTION
**Pull request by kshashov**

### CODE REVIEW

File: src/views/profile/ProfileForm.vue [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    },
+    watch: {
+      item: function () {
+        this.updateUser()
+      }
+    },
+    created () {
+      this.updateUser()
+    },
+    methods: {
```
Comment (maxsinev) [inline src/views/profile/ProfileForm.vue]:
```suggestion
    watch: {
      item: {
        handler() {
          this.updateUser()
        },
        immediate: true
      }
    },
    methods: {
```

File: src/store/index.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import Vuex from 'vuex'
+import { ACCESS_TOKEN } from '../constants'
+import { getCurrentUser } from '../utils/APIUtils'
+import router from '../router'
+import { snackbar } from './snackbar'
+import { langs } from './langs'
+import { tags } from './tags'
+import { roles } from './roles'
+import { exercisesStats } from './exercisesStats'
+import lodash from 'lodash'
```
Comment (maxsinev) [inline src/store/index.js]: https://stackoverflow.com/questions/35250500/correct-way-to-import-lodash

File: src/views/profile/ProfileForm.vue [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      updateUser () {
+        this.user = this.item
+          ? lodash.cloneDeep(this.item)
+          : undefined
+      },
+      save () {
+        this.$v.$touch()
+        if (!this.$v.$invalid) {
+          this.onSave(this.user)
```
Comment (maxsinev) [inline src/views/profile/ProfileForm.vue:75]: React style? why not `this.$emit("save", { user })` and in parent `<ProfileForm @save="onSave" ...>`

File: src/views/exercise/Exercise.vue [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+      canSolveExercise () {
+        return this.$can('solve', 'Exercise')
+      },
+      ...mapGetters(['authenticated'])
+    },
+    created () {
+      this.startLoading()
+    },
+    mounted () {
+      const self = this
```
Comment (maxsinev) [inline src/views/exercise/Exercise.vue]: redundant, `this` works in all callbacks below

**Review by maxsinev — COMMENTED** [review]
👍

File: src/views/profile/ProfileForm.vue (hunk 3 above)
Comment (kshashov) [reply src/views/profile/ProfileForm.vue:75]: I prefer to handle callbacks, at least required ones, as props, so I can be sure that I don't forget to pass them

**Pull request merged**